### ISSUE TITLE
order-dependent optimization, and batch sizes

### ISSUE THREAD

**Issue by adelmemariani** [dice-group/dice-embeddings#356]
I’m a bit concerned about potential order-dependent optimization effects when training with Adam or SGD.

For UMLS, with batch size 1024, learning rate 0.1, and DistMult, without "reciprical" triples I get the following:

`Evaluate DistMult on Test set: Evaluate DistMult on Test set
{'H@1': 0.7307110438729199, 'H@3': 0.8381240544629349, 'H@10': 0.9349470499243571, 'MRR': 0.8003702286494183}`

When I move the first 10 triples of the dataset to the end, I get:

`Evaluate DistMult on Test set: Evaluate DistMult on Test set
{'H@1': 0.6656580937972768, 'H@3': 0.8381240544629349, 'H@10': 0.9425113464447806, 'MRR': 0.7617573341100848}`

Here are the triples that I moved:
`carbohydrate	interacts_with	vitamin
molecular_sequence	isa	spatial_concept
steroid	affects	mental_process
experimental_model_of_disease	occurs_in	professional_or_occupational_group
patient_or_disabled_group	uses	clinical_drug
professional_or_occupational_group	produces	clinical_drug
nucleic_acid_nucleoside_or_nucleotide	isa	chemical_viewed_structurally
organism_function	degree_of	mental_process
steroid	affects	cell_or_molecular_dysfunction`

Therefore, I was wondering if it is possible do the gradient accumulation after each epoch (not steps).
